### ISSUE TITLE
AppBar accessibility traversal order 

### ISSUE THREAD

**Issue by yarmDev** [flutter/flutter#98570]
Hello, 

When we use a SliverAppBar, we notice that the semantic order is defined in this way: Leading -> Title -> Actions -> Bottom -> FlexingSpace.
https://github.com/flutter/flutter/blob/ee0bbf4789fac99576c0bd249466a20a305af606/packages/flutter/lib/src/material/app_bar.dart#L1027
I think it could be useful to be able to modify this order, for example the Bottom at the end

Thanks for your help :)

**Comment by darshankawar** [comment]
@yarmDev 
Check [this](https://github.com/flutter/flutter/issues/76108#issuecomment-780723834) issue comment for your reference and see by using `Semantics(
            sortKey: OrdinalSortKey` if it helps to achieve your use case.

**Comment by yarmDev** [comment]
I have already tried to use the sortKey in the SliverAppBar, it does not change.
I think the sortkeys inside the appBar take over

**Comment by darshankawar** [comment]
@yarmDev 
Can you provide a code sample that shows the current behavior along with an expected result you are looking for ?

**Comment by yarmDev** [comment]
```dart
    SliverAppBar(
      floating: false,
      pinned: true,
      snap: false,
      expandedHeight: getExpandedHeight(),
      leading: canPop(context)
          ? Container(
              margin: const EdgeInsets.all(Dimens.halfSpacing),
              child: AppBarBackButton(
                onPressed: pop(context),
              ))
          : null,
      actionsIconTheme: IconThemeData(
        color: Theme.of(context).colorScheme.onPrimary, //change your color here
      ),
      iconTheme: IconThemeData(
        color: Theme.of(context).colorScheme.onPrimary, //change your color here
      ),
      // status bar brightness
      elevation: 0,
      backgroundColor: Colors.transparent,
      title: AnimatedOpacity(
        duration: const Duration(milliseconds: 300),
        opacity: widget.showSmallBar ? 1.0 : 0.0,
        curve: Curves.ease,
        onEnd: () {
          setState(() {
            renderSmallBar = widget.showSmallBar;
          });
        },
        child: renderSmallBar ? widget.smallBar : null,
      ),
      collapsedHeight: widget.disableCollapse == true ? getExpandedHeight() : null,
      flexibleSpace: BackgroundFlexibleSpaceBar(
        collapseMode: CollapseMode.pin,
        background: Container(
          color: Theme.of(context).colorScheme.background,
          child: AnimatedOpacity(
            duration: const Duration(milliseconds: 300),
            opacity: !widget.showSmallBar ? 1.0 : 0.0,
            curve: Curves.ease,
            onEnd: () {
              setState(() {
                renderSmallBar = widget.showSmallBar;
              });
            },
            child: widget.expandedBar,
          ),
        ),
      ),
      bottom: widget.bottom,
    );
```
this is the current behavior:




https://user-images.githubusercontent.com/51370782/154477040-1b52a417-67b6-4aee-a850-cf3986588b19.mov

I would like to be able to select the searchForm after the backButton, it seems more logical to me

**Comment by darshankawar** [comment]
@yarmDev 
Have you tried using `ExcludeSemantics` widget ? You can read more about it here https://api.flutter.dev/flutter/widgets/ExcludeSemantics-class.html and see if it helps to arrange traversal order per your requirement.

**Comment by yarmDev** [comment]
In fact I wouldn't want to exclude the bottom.

**Comment by darshankawar** [comment]
@yarmDev 
It seems the code you shared has lot of custom implementation, can you provide a runnable code sample that we can directly run and verify ?

**Comment by yarmDev** [comment]
```dart
SliverAppBar(
      leading: Text('BackButton'),
      flexibleSpace: BackgroundFlexibleSpaceBar(
        collapseMode: CollapseMode.pin,
        background: Container(
          child: Center(child: Container(child: const Text('FlexibleSpace')),
          ),
        ),
      ),
      bottom: PreferredSize(
          preferredSize: const Size.fromHeight(64),
          child: Container(
            child: Text('Tabs'),
          )),
    );

**Comment by darshankawar** [comment]
Thanks for the updated code. Looking at it, I believe you are hitting https://github.com/flutter/flutter/issues/77486, so please follow-up in it for further updates.
Closing this as a duplicate. If you disagree, write in comments and I'll reopen it.

**Comment by darshankawar** [comment]
Reopening based on https://github.com/flutter/flutter/issues/77486#issuecomment-1062935996

**Comment by yarmDev** [comment]
Hello, I would like the "SliverAppBar" from flexibleSpace to be read by TalkBack just after the "BackButton"

**Comment by darshankawar** [comment]
Thanks. I am keeping this open and labeling for further input from the team.

/cc @chunhtai

**Comment by GODOM018** [comment]
Hello!
I can also reproduce it with an `AppBar`. It does not seem to honor the `OrdinalSortKey` given. 

Flutter 3.27.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 17025dd882 (3 weeks ago) • 2024-12-17 03:23:09 +0900
Engine • revision cb4b5fff73
Tools • Dart 3.6.0 • DevTools 2.40.2

```
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(
    MaterialApp(
      initialRoute: 'home',
      routes: {
        'home': (context) => const HomePage(),
        'test': (context) => const TestPage(),
      },
    ),
  );
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: ElevatedButton(
          style: ButtonStyle(
            backgroundColor: WidgetStatePropertyAll(Colors.pink),
          ),
          onPressed: () => Navigator.pushNamed(context, 'test'),
          child: Text(
            'Navigate to test page',
            style: TextStyle(
              color: Colors.white,
            ),
          ),
        ),
      ),
    );
  }
}

class TestPage extends StatelessWidget {
  const TestPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Colors.grey,
        centerTitle: true,
        leading: Semantics(
          sortKey: OrdinalSortKey(2),
          child: BackButton(
            color: Colors.black,
          ),
        ),
        title: Semantics(
          sortKey: OrdinalSortKey(1),
          child: Text('app bar title'),
        ),
      ),
      body: Center(
        child: Text('Hello World!'),
      ),
    );
  }
}
```

**Comment by chunhtai** [comment]
Looks like it is the sort key in the appbar that causes issue, we will probably need to investigate how to properly assign sortkey.

**Comment by MuneerAhamed01** [comment]
For a quick fix remove disable the **respective widgets headers as false** @GODOM018 

@yarmDev  Can you please try it in the SliverAppBar Also

AppBar(
          title: Semantics(
            sortKey: OrdinalSortKey(1),
            headingLevel: 1,
            child: Text(
              'User Registration',
              semanticsLabel: 'User Registration Page',
            ),
          ),
          centerTitle: true,
          leading: Semantics(
            sortKey: OrdinalSortKey(2),
            header: false,
            // label: 'Go back',
            child: BackButton(),
          ),
        ),
